protocol/strict-derivation.md

```diff
@@ -0,0 +1,269 @@
+# Purpose
+
+Holocene aims to introduce a set of changes to derivation that improve worst-case scenarios for
+Fault Proofs and Interop. The goal of this design doc is to align on the changes to derivation that
+will be part of the upcoming Holocene hardfork, and in particular to solve open questions to
+finalize the specs.
+
+The Holocene specs for these derivation changes are currently being written in [specs/#357](https://github.com/ethereum-optimism/specs/pull/357).
+
+# Summary
+
+The Holocene hardfork introduces several changes to batch derivation:
+
+- _Partial Span Batch Validity_ determines the validity of singular batches within a span batch
+individually, only invalidating the remaining span batch upon the first invalid batch.
+- _Steady Batch Derivation_ derives invalid singluar batches immediately as deposit-only
+blocks.
+- _Strict Batch Ordering_ requires batches within and across channels to be strictly ordered.
+
+The design space and some proposed solutions will be discussed, together with pratical implications
+for batcher implementations that have to satisfy the stricter rules.
+
+# Partial Span Batch Validity
+
+## Problem Statement + Context
+
+Before Holocene, Span Batch validity is handled atomically, i.e., either all singluar batches that
+can be derived from a span batch are valid, or if any of them is invalid (during batch queue
+checks), the whole span batch is dropped. This has negative implications for Fault Proofs: in a
+worst-case scenario, if the to-be-fault-proven block is the first batch in a span batch, to
+determine its validity at the batch queue stage, the validity of the full span batch has to be
+determined, as any later invalid singular batch would also invalidate all prior batches in the same
+span batch. So only invalidating the remaining span batch upon hitting an invalid derived singular
+batch vastly improves this scenario, since batch queue validity is then final for any singular
+batch, not just regular singular batches.
+
+In late 2023 while Delta was still in development, the problem of partial span batch validity was
+[already discussed](https://www.notion.so/oplabs/Handling-Invalid-Batches-48bb972368ce409c84bd9189eddd0577)
+and for simplicity it was decided to use the atomic validity model. However, above discussion
+focusses on implications of the atomic validity model for EL-sync. We're now concerned with behavior
+around CL client restarts and reorgs.
+
+### Mid-way restart
```

Comment:
To me it would be more natural to put this section below the "Proposed Design" section, especially since it apparently raises a potential issue but then dismisses it.

Reply:
fixed in https://github.com/ethereum-optimism/design-docs/pull/73 right?

Reply:
I added it here because it first explains a problem, and then I just gave my reasoning as to why I believe it isn't an issue, but I don't know if it's the correct answer. I will consider moving it around.